### PR TITLE
fix : empty password or email bug

### LINKED ISSUE #37: app crashes if username and password field is empty
empty username or empty password causes the app to crash
working on it 👍

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #37

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt [hunk 1]
```diff
@@ -10,6 +10,7 @@ class AuthService(private val authApi: AuthApi,
 
     private val userMap = ConcurrentHashMap<Long, User>() // TODO: To be replaced by room
 
+    @Throws(IllegalArgumentException::class)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt]: Checked exceptions are very bad

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginActivityViewModel.kt [hunk 2]
```diff
@@ -18,18 +18,22 @@ class LoginActivityViewModel(private val authService: AuthService) : ViewModel()
...
+                   }.doFinally {
+                       progress.value = false
+                   }.subscribe({
+                       loggedIn.value = true
+                   }, {
+                       error.value = "Unable to Login. Please check your credentials"
+                   }))
+       } catch (e : IllegalArgumentException) {
+           e.printStackTrace()
+       }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginActivityViewModel.kt]: Umm.. Rather add a check and send error to view?

**Comment by simarsingh24** [comment]
I am adding simple empty check and a toast message,would that be fine?

**Comment by iamareebjamal** [comment]
Yes that is what should be done

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginActivity.kt [hunk 3]
```diff
@@ -24,7 +24,11 @@ class LoginActivity : AppCompatActivity() {
         setContentView(R.layout.activity_login)
 
         loginButton.setOnClickListener {
-            loginActivityViewModel.login(username.text.toString(), password.text.toString())
+            if (username.text.toString().isEmpty() || password.text.toString().isEmpty()) {
+                Toast.makeText(this, "Username or Password can not be empty!", Toast.LENGTH_SHORT).show()
+            } else {
+                loginActivityViewModel.login(username.text.toString(), password.text.toString())
+            }
```
[the remark this answers is not in the document]
Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginActivity.kt]: ohkay

[the remark this answers is not in the document]
Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginActivity.kt]: @iamareebjamal To display toast I need context and I cant pass context to a view model, So how should I do it. Should I send back a boolean from view model and display toast from activity by checking the value of boolean?

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginActivity.kt]: Look at how it is done. This is already implemented in this LoginActivityViewModel already

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginActivityViewModel.kt [hunk 4]
```diff
@@ -18,6 +18,10 @@ class LoginActivityViewModel(private val authService: AuthService) : ViewModel()
     fun isLoggedIn() = authService.isLoggedIn()
 
     fun login(email: String, password: String) {
+        if (email.isNullOrEmpty() || password.isNullOrEmpty()) {
+            progress.value = false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginActivityViewModel.kt]: Progress would be false already AFAIK

Send the error to view. "Email and Password cannot be empty"

**Pull request merged**